packages/nodes-base/nodes/Code/Sandbox.ts

```diff
@@ -191,70 +143,68 @@ export class Sandbox extends NodeVM {
 			});
 		}
 
-		if (executionResult.json !== undefined && !isObject(executionResult.json)) {
+		if (Array.isArray(executionResult)) {
+			const firstSentence =
+				executionResult.length > 0
+					? `An array of ${typeof executionResult[0]}s was returned.`
+					: 'An empty array was returned.';
+
+			throw new ValidationError({
+				message: "Code doesn't return a single object",
+				description: `${firstSentence} If you need to output multiple items, please use the 'Run Once for All Items' mode instead`,
+				itemIndex: this.itemIndex,
+			});
+		}
+
+		const [returnData] = this.helpers.normalizeItems([executionResult]);
```

Comment:
This like fixes ADO-513, by normalizing the return-date before validating the keys.